src/server/game/Globals/ObjectMgr.cpp

```diff
@@ -2469,6 +2469,74 @@ uint32 ObjectMgr::AddCreData(uint32 entry, uint32 mapId, float x, float y, float
     return spawnId;
 }
 
+void ObjectMgr::UpdateCreatureHalaa(ObjectGuid::LowType spawnId, TeamId teamId, Map* map)
```

Comment:
Thats why i ask, should be better all NPCS hardcoded or still continue investigating? because i don't like too much this but dunno how to make it through DB (Tried with 40 mobs but the game i saw doesnt spawn more than 20 - 30?), so asking what do you preffer and i can change

Reply:
> This is just a hack with ugly hardcoded stuff

Just tried again with the 40 mobs and they are spawning, before i wasn't able to spawn the 40 mobs because some of them were spawning dead (Probably the ACAdmin addon was killing someone), gonna try again with all of them spawned and only changing phasemask.

Reply:
First of all, they should all be present with their correct values in the `creature` table.
Then I think I would suggest writing a script for spawning/despawning the creatures depending on the WorldState which is updated on takeover

Reply:
> First of all, they should all be present with their correct values in the `creature` table. Then I think I would suggest writing a creature script for spawning/despawning the creatures depending on the WorldState which is updated on takeover

Hmm, thx for the tip, last time i was using creatureScript but dunno how to use, now i saw something and prolly i can get something better than this and i can delete that Hardcoded stuff :D

Reply:
Changed but im not convinced in script, i still don't know how CreatureScripts works, tried with all creatures with scriptname : OutdoorPvP_nagrandCreature but i got a list error and using this "hack" i saw there is a function called OnCreatureAddWorld and OnCreatureCreate but code never get inside there